cmd/namespace-lock.go

```diff
@@ -120,8 +112,28 @@ type nsParam struct {
 
 // nsLock - provides primitives for locking critical namespace regions.
 type nsLock struct {
-	RWLockerSync
-	ref uint
+	DRWMutexLock *dsync.DRWMutex
+	LRWMutexLock *lsync.LRWMutex
+	ref          uint
+}
+
+func (n *nsLock) Unlock() {
+	if n.DRWMutexLock != nil {
+		n.DRWMutexLock.Unlock()
+	}
+	if n.LRWMutexLock != nil {
+		n.LRWMutexLock.Unlock()
```

Comment:
Can you update this PR to using latest lsync? with `go.mod` ?

Reply:
done